src/modules/licence-import/load/licence.js

```diff
@@ -45,15 +45,18 @@ const loadAgreements = async licence => {
 };
 
 const loadPurposeConditions = (purposeId, purpose) => {
-  return Promise.all(purpose.conditions.map(async condition => {
+  Promise.all(purpose.conditions.map(async condition => {
     return connectors.createPurposeCondition(condition, purposeId);
   }));
 };
 
 const loadVersionPurposes = (licenceVersionId, version) => {
   return Promise.all(version.purposes.map(async purpose => {
     const purposeResult = await connectors.createLicenceVersionPurpose(purpose, licenceVersionId);
-    return loadPurposeConditions(purposeResult.licence_version_purpose_id, purpose);
+    if (purpose.conditions.length > 0) {
```

Comment:
Not too sure why this is needed either, because Promise.all() should be happy with an empty array.

That's unless purpose.conditions is sometimes not an array.

In this case it may be clearer to have:

```
const {isArray} = require('lodash');

// ...

if(isArray(purpose.conditions)) {
  // ....